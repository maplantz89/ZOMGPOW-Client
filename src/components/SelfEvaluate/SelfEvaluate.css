.self-evaluate-form{
  display: flex; 
  flex-direction: column;
  padding: 15px;
  border-radius: 15px;
  margin: 0 auto;
  width: 80%;
  max-width: 300px;
  align-items: center;
}

.option .low {
  background: white;
  color: var(--purple);
  border-radius: 3px;
  margin-left: 10px;
  vertical-align: middle;
  padding: 15px;
}

.option .medium {
  background: white;
  color: var(--green);
  border-radius: 3px;
  margin-left: 10px;
  vertical-align: middle;
  padding: 15px;
}

.option .high {
  background: white;
  color: var(--blue);
  border-radius: 3px;
  margin-left: 10px;
  vertical-align: middle;
  padding: 15px;
}
