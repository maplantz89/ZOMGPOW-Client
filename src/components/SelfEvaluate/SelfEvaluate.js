import React from 'react';
import './SelfEvaluate.css';
import StudentAuthApiService from '../../Services/student-auth-api-service';

class SelfEvaluate extends React.Component {
  state = {
    score: null,
    error: null,
  }

  setScore=(e) => {
    this.setState({
      score: e.target.value,
    })
  }

  postEvaluation = (e) => {
    e.preventDefault()
    const studentScore = this.state.score;
    const student_goal_id = this.props.location.state.learningTarget.sg_id;
    if(this.props.location.state.currentGoal === undefined){
      StudentAuthApiService.patchStudentGoal(student_goal_id, {evaluation: studentScore})
      .then(() => this.props.history.goBack())
      .catch(error => {
        console.error(error);
        this.setState({ error })
      });
    }
    else{
    const id = this.props.location.state.currentGoal.id;
      StudentAuthApiService.patchSubGoal(id, {evaluation: studentScore})
      .then(() => this.props.history.goBack())
      .catch(error => {
        console.error(error);
        this.setState({ error })
      });
    }
  }
  
  render(){
    return (
      <div className='self-evaluate-form'>
      <form className='form' onSubmit={this.postEvaluation}>
      <h3>Rate your level of understanding:</h3>
        <input
          onChange={(e) => this.setState({ score: 1 })}
          className='radio'
          type='radio'
          value='low'
          id='low'
          name='priority'
          required/>
        <label
          htmlFor='low'>Thumbs down</label>
        <input
         onChange={(e) => this.setState({ score: 2 })}
          className='radio'
          type='radio'
          value='medium'
          id='medium'
          name='priority'/>
        <label
          htmlFor='medium'>Thumb sideways</label>
        <input
         onChange={(e) => this.setState({ score: 3 })}
          className='radio'
          type='radio'
          value='high'
          id='high'
          name='priority'/>
        <label
          htmlFor='high'>Thumbs up</label>
        <button className='button purple-button'>Submit Evaluation</button>
      </form>
      </div>
    )
  }
}

export default SelfEvaluate;