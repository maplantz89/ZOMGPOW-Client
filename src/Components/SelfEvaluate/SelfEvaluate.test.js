import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SelfEvaluate from './SelfEvaluate';

describe(`SelfEvaluate component`, () => {
  it('renders a .SelfEvaluate by default', () => {
    const wrapper = shallow(<SelfEvaluate />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})