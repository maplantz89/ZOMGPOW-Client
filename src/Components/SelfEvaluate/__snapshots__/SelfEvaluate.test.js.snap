// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelfEvaluate component renders a .SelfEvaluate by default 1`] = `
<div
  className="self-evaluate-form"
>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <h3>
      Rate your level of understanding:
    </h3>
    <div
      className="option"
    >
      <input
        className="radio"
        id="high"
        name="priority"
        onChange={[Function]}
        type="radio"
        value="high"
      />
      <label
        htmlFor="high"
      >
        <FontAwesomeIcon
          border={false}
          className="high"
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "far",
              "smile",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="2x"
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </label>
    </div>
    <div
      className="option"
    >
      <input
        className="radio"
        id="medium"
        name="priority"
        onChange={[Function]}
        type="radio"
        value="medium"
      />
      <label
        htmlFor="medium"
      >
        <FontAwesomeIcon
          border={false}
          className="medium"
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "far",
              "meh",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="2x"
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </label>
    </div>
    <div
      className="option"
    >
      <input
        className="radio"
        id="low"
        name="priority"
        onChange={[Function]}
        required={true}
        type="radio"
        value="low"
      />
      <label
        htmlFor="low"
      >
        <FontAwesomeIcon
          border={false}
          className="low"
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "far",
              "angry",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="2x"
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </label>
    </div>
    <button
      className="button purple-button"
    >
      Submit Evaluation
    </button>
  </form>
</div>
`;
